src/src/com/microsoft/aad/adal/StorageHelper.java

```diff
@@ -369,34 +367,31 @@ private synchronized SecretKey getKeyOrCreate(final String keyVersion) throws Ge
      * @throws IOException
      */
     private synchronized SecretKey getKey(final String keyVersion) throws GeneralSecurityException, IOException {
-        if (VERSION_USER_DEFINED.equals(keyVersion)) {
+        switch (keyVersion) {
+        case VERSION_USER_DEFINED : 
             return getSecretKey(AuthenticationSettings.INSTANCE.getSecretKeyData());
-        }
-        
-        if (VERSION_ANDROID_KEY_STORE.equals(keyVersion)) {
+        case VERSION_ANDROID_KEY_STORE :
             if (Build.VERSION.SDK_INT >= 18) {
```

Comment:
@weijjia I thought that we decided we didn't need this check.  We should never have a case where there is a key in the key store on a device with an OS below 18.